ray-operator/controllers/ray/rayjob_controller.go

```diff
@@ -897,6 +902,30 @@ func checkK8sJobAndUpdateStatusIfNeeded(ctx context.Context, rayJob *rayv1.RayJo
 			return true
 		}
 	}
+	if rayv1.IsJobTerminal(rayJob.Status.JobStatus) && rayJob.Status.JobDeploymentStatus == rayv1.JobDeploymentStatusRunning {
+		submitterGracePeriodTime, err := strconv.Atoi(os.Getenv(utils.SUBMITTER_GRACE_PERIOD_TIME))
+		if err != nil {
+			logger.Info(
+				"Environment variable is not set, using default value of seconds",
+				"environmentVariable", utils.SUBMITTER_GRACE_PERIOD_TIME,
+				"defaultValue", utils.SUBMITTER_DEFAULT_GRACE_PERIOD_TIME,
+			)
+			submitterGracePeriodTime = utils.SUBMITTER_DEFAULT_GRACE_PERIOD_TIME
+		}
+
+		if time.Now().Before(rayJob.Status.StartTime.Add(time.Duration(submitterGracePeriodTime) * time.Second)) {
```

Comment:
we should use the time that JobStatus transitions to a terminal state instead of `StartTime`.